### PR TITLE
[NCL-5558] Respond with 403 status when Brew Push is forbidden becaus…

### PR DESCRIPTION
**Pull request by jomrazek**
…e of bad artifact quality

### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by jomrazek** [comment]
retest this please

File: facade/pom.xml [hunk 1]
```diff
@@ -139,6 +139,11 @@
             <artifactId>hibernate-core</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-jaxrs</artifactId>
+            <scope>provided</scope>
+        </dependency>
```
Comment (janinko) [inline facade/pom.xml]: Why is this needed?

File: facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java [hunk 2]
```diff
@@ -121,6 +119,10 @@ public BuildPushResult brewPush(String id, BuildPushRequest buildPushRequest) th
                                 .build())
                 .collect(Collectors.toList());
 
+        if (pushed.stream().anyMatch(r -> r.getMessage().contains("BLACKLISTED/DELETED"))) {
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java]: use the `pushedResponse.get(0)` before this if and everything will be much simpler.

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildEndpoint.java [hunk 3]
```diff
@@ -328,6 +330,10 @@ void removeAttribute(
                             responseCode = INVALID_CODE,
                             description = INVALID_DESCRIPTION,
                             content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = FORBIDDEN_CODE,
+                            description = FORBIDDEN_PUSH_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
```
Comment (janinko) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildEndpoint.java]: the schema is actually BuildPushResultPage

File: facade/pom.xml (hunk 1 above)
Comment (jomrazek) [reply facade/pom.xml]: ClientErrorException

Comment (michalovjan) [reply facade/pom.xml]: Maybe throw different exception. Wrap it in the rest-new module or write exception mapper?

File: rest-api/src/main/java/org/jboss/pnc/rest/configuration/SwaggerConstants.java [hunk 4]
```diff
@@ -42,6 +42,7 @@
     public static final String INVALID_CODE = "400";
     public static final String FORBIDDEN_DESCRIPTION = "User must be logged in.";
     public static final String FORBIDDEN_CODE = "403";
+    public static final String FORBIDDEN_PUSH_DESCRIPTION = "Build contains artifacts of of insufficient quality";
```
[the remark this answers is not in the document]
Comment (jomrazek) [reply rest-api/src/main/java/org/jboss/pnc/rest/configuration/SwaggerConstants.java]: eagle eye!

**Comment by jomrazek** [comment]
retest this please

File: facade/pom.xml (hunk 1 above)
Comment (janinko) [reply facade/pom.xml]: ClientErrorException should be in some javax.ws module (rs-api?)

Comment (jomrazek) [reply facade/pom.xml]: any new dependency is completely removed in most recent version of PR

Comment (michalovjan) [reply facade/pom.xml]: @janinko WDYT? I am fine, with the changes.
